src/iosrv/srv.c

```diff
@@ -261,6 +341,7 @@ dss_sched_run(ABT_sched sched)
 
 			ABT_sched_has_to_stop(sched, &stop);
 			if (stop == ABT_TRUE) {
+				ds_check_expire_list(true);
```

Comment:
Pleases see dss_srv_handler(), we wait for all other ULTs exiting when exiting progress ULT, so I still don't see why forcing wakup here work.

Reply:
indeed, we should stop the ult being added to the timeout list again here.